### ISSUE TITLE
x265 --atc-sei and --pic-struct wrong type

### ISSUE THREAD

**Issue by tkozybski** [staxrip/staxrip#593]
Those option take an integer (not signaled by default), They are not a flag/boolean.

Also: It's just a wording issue, but SAR defaults to undefined, not automatic.

**Comment by stax76** [comment]
Auto is correct, in the main dialog there is a target PAR menu, change it to something different from 1:1, and you will see that value in the encoder command line.

**Comment by tkozybski** [comment]
> 
> 
> Auto is correct, in the main dialog there is a target PAR menu, change it to something different from 1:1, and you will see that value in the encoder command line.

Neat, didn't see that before,. But If I set 1:1 on the main menu, it won't set 1:1 in the command line.

**Comment by stax76** [comment]
That's by design and this is a good design.

In the encoder dialog you can change it from auto to 1:1 using the menu. But you might need a recent beta for this to work since there was a fix some time ago, I don't remember when exactly.

**Comment by Dendraspis** [comment]
I've fixed it.

Hotfix:
[StaxRip v2.1.8.5D_x265.zip](https://github.com/staxrip/staxrip/files/6141994/StaxRip.v2.1.8.5D_x265.zip)

**Comment by tkozybski** [comment]
> 
> 
> I've fixed it.
> 
> Hotfix:
> [StaxRip v2.1.8.5D_x265.zip](https://github.com/staxrip/staxrip/files/6141994/StaxRip.v2.1.8.5D_x265.zip)

Thank you, but the GUI looks broken now because of the long description.

**Comment by Dendraspis** [comment]
You're right.  😂 

Take this out, looks a bit better:  😁 
[StaxRip v2.1.8.5D_x2652.zip](https://github.com/staxrip/staxrip/files/6142111/StaxRip.v2.1.8.5D_x2652.zip)

**Comment by stax76** [comment]
I think Master Display is too small with that, I committed something possible better, but it still needs some reordering.

**Comment by tkozybski** [comment]
> 
> 
> That's by design and this is a simple and good design if you think about it, internally an undefined state does not exist, it's just a numeric value.
> 
> In the encoder dialog you can change it from auto to 1:1 using the menu. But you might need a recent beta for this to work since there was a fix some time ago, I don't remember when exactly.

I don't think I follow. Currently if you set sar to auto, it will not set it in the encoder if it's 1:1, --sar won't be used which defaults to undefined (not signaled), and the decoder decides how to scale pixels. Auto is not really auto if it doesn't signal anything :)

My understanding of that parameter is that there should be three types of settings:
- Undefined/Not signaled
- Auto (set it based on PAR, even if it's 1:1)
- All other manual settings

**Comment by stax76** [comment]
I might have an idea, but I'm going for a walk first.

**Comment by stax76** [comment]
Maybe a new project option:

Set PAR even if it's 1:1

**Comment by Dendraspis** [comment]
@stax76 
I don't think this is better:  😉 

<img src="https://user-images.githubusercontent.com/50659978/111184762-9460a780-85b1-11eb-8aba-fc180a73fa36.PNG">

**Comment by stax76** [comment]
'Alt. transfer characteristics SEI' is long, so Master Display becomes short, possible change could be:

'Alt. transfer char. SEI'

Or putting the num edit below the label for only this option.

Other suggestions:

'Picture structure in SEI:' -> 'Pic structure in SEI:'

Pic is a common shortcut.

'RSkip Edge Threshold' -> 'RSkip Edge Thres.'

'Maximum Picture Order Count' -> 'Max Pic Order Count'

I can only suggest it, maybe @44vince44 or @JJKylee have an opinion or idea.

**Comment by 44vince44** [comment]
Well there is no way at all to put the full text and to move buttons to the right ? I mean, layout is important, but displaying clear information is more important....

'Alt. transfer char. SEI' is really not easy to understand 😕

**Comment by stax76** [comment]
@44vince44 

Did you see my solution where the label is on top of the num edit?

https://github.com/staxrip/staxrip/issues/593#issuecomment-799545940

The aspect ratio thing is implemented, I'm writing now an explanation.

**Comment by JJKylee** [comment]
I like @stax76's idea of abbreviating the terms.
FYI, 'transfer characteristics' can be shortened just to 'transfer'. 'characteristics' is usually omitted in many cases.

And about `PAR`, I agree with @stax76 and @44vince44. 😄

**Comment by Dendraspis** [comment]
Okay, you guys are still talking - I thought this is done.

**Comment by tkozybski** [comment]
Tooltips on mouse hover could explain the settings more precisely, even contain the description from docs. This way shorter names could be used and still provide all the information needed.

**Comment by stax76** [comment]
I don't think we will support tooltip, but context help exist via right-click, this is documented, press F1.

**Comment by tkozybski** [comment]
The more you know... I don't see any issues with shorter names then.

**Comment by Dendraspis** [comment]
My suggestion:  Make that a bit window wider.   😁 

But I'll wait 'till you come out with a decision. I don't understand this discussion.  🙄

**Comment by stax76** [comment]
JJKylee is right, characteristics can be omitted like it is for --transfer, I've changed it and increased the Master Display height to 2 lines.

> My suggestion: Make that window a bit wider.

I increased it from 38 to 40, --rd fits now better, but still I think it's better when labels are short.

The aspect ratio improvement:

The PAR menu is used in three locations, 2 times in the main dialog and 1 time in the encoder dialog.

In the PAR menu there is now a new option 'Forced' but this will only show up when the menu is manually reset, this can be done in the menu editor in the main dialog.

In all three PAR menus forced can be set, this will set --sar in the encoder even if PAR is 1:1.

In the main dialogs target PAR menu setting to 1:1 will also force it in the encoder.

Hotfix is online.
